### ISSUE TITLE
Empty Code Time Panel and not able to lig in

### ISSUE THREAD

**Issue by anandinasu** [swdotcom/swdc-vscode#324]
**Describe the bug**
From [here](https://marketplace.visualstudio.com/items?itemName=softwaredotcom.swdc-vscode), it is clear that the login is done through VS Code. But unfortunately, my Code Time Sidebar is empty. Literally nothing there. I re installed and it is still the same. A blank space.

**ISSUE TYPE**
- Bug Report

**Steps to reproduce**
1. Install the extension

**Expected behavior**
Should show a "Getting Started" prompt

**ACTUAL RESULTS**
Blank space

**Screenshots**
![image_108](https://user-images.githubusercontent.com/12847193/132517048-f587aacd-5010-4ae4-a4fd-828c786701a0.png)

**Desktop (please complete the following information):**
 - Windows 10
 - VS Code v1.60
 - Plugin version Code Time v2.5.24

**Comment by mmclead** [comment]
Thank you for reporting @anandinas.

We'll start looking at that right away.

**Comment by xavluiz** [comment]
We're looking into this now on Windows, but currently unable to reproduce the issue. Would you be able to check the console within the VSCode developer tools to see if there are any CodeTime errors preventing the display of the webview? The console is available within the Help->Toggle Developer Tools menu.
Thank you.

**Comment by anandinasu** [comment]
Thank you for the response :)

> We're looking into this now on Windows, but currently unable to reproduce the issue. Would you be able to check the console within the VSCode developer tools to see if there are any CodeTime errors preventing the display of the webview? The console is available within the Help->Toggle Developer Tools menu.
> Thank you.

![image_110](https://user-images.githubusercontent.com/12847193/132627874-f40ce2e4-09b4-464b-883c-6ac9e9920093.png)

**Comment by anandinasu** [comment]
Guys, I understood the problem here and fixed it.

TL;DR; Moved default location of extension folder to OneDrive which has path length over 256. Moving it back fixed the issue.

I had moved my .vscode folder from default location ("C:\Users\UserName\.vscode\") to my OneDrive folder with the help of Symlink Directory Junction. Apart from that, it is worth mentioning that the path length to the root folder of your extension is 209 and the longest path length is 263 (".\extensions\softwaredotcom.swdc-vscode-2.5.24\dist\src\message_handlers\AUTHEN~1.JS"). 

So I tried moving back to the default folder ("C:\Users\UserName\.vscode\") and now I can see the dashboard. I have successfully logged in.

I am not sure what exactly caused the issue. Whether the Symlink or the path length or something else. If I may ask, have you guys hard-coded any location? 

Very sorry to bother you about this and thanks a lot for your time.

**Comment by xavluiz** [comment]
Thank you very much for this information and it's no bother at all. The path length is important and we'll look into our extension ID naming convention to see if there's something we can do to shorten our ID in that path. In regards to hard-coded location, the plugin writes a json file into the home directory under the a directory called .software in order to maintain the session across code time editors.  Thanks again for your help in figuring out this issue.
Regards.

**Comment by anandinasu** [comment]
> The path length is important and we'll look into our extension ID naming convention to see if there's something we can do to shorten our ID in that path

Haha you are way too polite. The extensions folder name is "softwaredotcom.swdc-vscode-2.5.24" which is 33 characters. Means out of 209, 176 characters is because of my "I-can-live-for-three-days-without-food-but-I-cant-survive-an-hour-with-an-unorganized-folder-structure" attitude. So that's my problem. You don't even have to think about changing it. However, now that we are talking about it, I wish the folder had the words "Code Time" in it  for easy recognition as I and probably others remember "Code Time" more than "softwaredotcom.swdc". 


> In regards to hard-coded location, the plugin writes a json file into the home directory under the a directory called .software in order to maintain the session across code time editors.

I understand. But I don't think you have hard-coded it as you go for "%USERPROFILE%/.software" folder. Otherwise, it would be impossible right. Oh, wait. Unless, you are looking for the parent folder location of where extension got saved (.vscode location) and search for a folder .software under that location. Then yeah it makes sense. If that is the case then if I still follow this hierarchy in the OneDrive folder, then it should work right? Unless the extension specifically rechecks whether the parent folder contains a folder strictly named ".vscode" to avoid any other issues. Logically speaking this makes more sense than the path length issue. I can recheck this couple of days later (bit busy now and also moving 20K files (entire .vscode folder) here and there is making OneDrive sync go crazy) and will update you, unless you know if this is how the extension works.

Thanks a lot 👍

**Comment by anandinasu** [comment]
Dear @xavluiz @mmclead,

> I can recheck this couple of days later

Like I mentioned in the previous post, I rechecked it partially (Did not do the "_follow this hierarchy in the OneDrive folder_" part). I realized that instead of moving all the files here and there, I could just download the portable version, follow the instructions mentioned [here](https://code.visualstudio.com/docs/editor/portable#_migrate-to-portable-mode) to replicate what I mentioned before and when I do this, I am getting an empty Code Time panel again like before. So it looks like, not sure though, what I mentioned in the previous post "_Unless, you are looking for the parent folder_" is true. 

This means that the people who are using a portable version of VScode will not be able to use your great extension. Unless the VScode team strongly opposes using the portable version (which they don't because they would not have made a portable version otherwise and also [here](https://github.com/microsoft/vscode/issues/329)), there will be people using that as the portable version has few more advantages and hence, it would be better if you find an alternative.

Another extension that uses a dedicated tab like Code Time which also works in portable mode (I rechecked it) is [Todo Tree](https://marketplace.visualstudio.com/items?itemName=Gruntfuggly.todo-tree). Maybe you can check how that works. I am not comparing your extension with theirs in any way. Just giving some information I found from my "testing".

Thank you.

**Comment by anandinasu** [comment]
Right after I posted the above post, when I went back to VSCode, a Code Time notification popped up saying it is the end of the day blah blah and when I clicked the view dashboard button, it showed the dashboard with metrics but the code time panel is still empty.

![image](https://user-images.githubusercontent.com/12847193/133057987-3ee5fad1-ee9a-496c-9af9-5aae63157a3e.png)

This confirms that the ".software" folder has got nothing to do with this problem.

**Comment by xavluiz** [comment]
hi @anandinasu,

I apologize for the late reply. I'll check into this and you're right, it should display the sidebar since the dashboard view shows the downloaded html. It is using React and there is specific file handling for a file called 'webviewSidebar.js'. I will look into this now and get back to you. Thanks again.

**Comment by Mwita04** [comment]
Thank you @anandinasu for bringing this up and offering the solution.

**Comment by xavluiz** [comment]
We're currently working on a web view solution for this and should have this out within a week. Thank you for your patience.
